cmd/snap/cmd_snap_op_test.go

```diff
@@ -1710,12 +1710,27 @@ func (s *SnapSuite) TestRefreshHoldBadDuration(c *check.C) {
 	})
 
 	rest, err := snap.Parser(snap.Client()).ParseArgs([]string{"refresh", "--hold=1d"})
-	c.Assert(err, check.ErrorMatches, "hold value must be a number of hours or minutes \\(e\\.g\\., 72h\\): time: unknown unit \"?d\"? in duration \"?1d\"?")
+	c.Assert(err, check.ErrorMatches, "hold value must be a number of hours, minutes or seconds, or \"forever\": time: unknown unit \"?d\"? in duration \"?1d\"?")
 	c.Assert(rest, check.DeepEquals, []string{"--hold=1d"})
 	c.Check(s.Stdout(), check.Equals, "")
 	c.Check(s.Stderr(), check.Equals, "")
 }
 
+func (s *SnapSuite) TestRefreshHoldNegativeDuration(c *check.C) {
+	s.RedirectClientToTestServer(func(w http.ResponseWriter, r *http.Request) {
+		c.Errorf("unexpected request")
+		fmt.Fprintln(w, `{"type": "error", "result": {"message": "received too many requests"}, "status-code": 500}`)
+	})
+
+	for _, dur := range []string{"-5h", "15ns"} {
```

Comment:
Super nitpick: maybe add "999ms" and a postivive test for "1s" (I assume we have a table driven test for accepted numbers.